### ISSUE TITLE
./alexorso/are-we-helping-developers.md

### ISSUE THREAD

**Issue by timm** [ds4se/chapters#21]
After review, relabel to 'reviewTwo'. After second review, relabel to 'EditorsComment'.

**Comment by murphygc** [comment]
## Title of chapter

_Are we really helping developers?_

## URL to the chapter

https://github.com/ds4se/chapters/blob/master/alexorso/are-we-helping-developers.md

## Message?

What is the chapter's  clear and approachable take away message?

_We can only determine if techniques intended to improve software development meet their intended goal(s) when we put the techniques and tools into the hands of users._

## Accessible?

Is the chapters written for a generalist audience (no excessive use of technical terminology) with a minimum of diagrams and references?
How can it be made more accessible to generalist?

_The chapter is accessible. The first header of "With Big Data Comes Great Responsibility" didn't quite align with the paragraphs that follow. It isn't clear to a general reader that a lot of data is the problem here, more that the research trajectory has been to analyze improvements in the absence of the humans who will use the technique._

_RQ1 needs some slight rewording (there is an extra 'will'). It is not clear such specific RQ formalism is required to get the point across._

_Typo: stageatistical :)_

## Size?

Is the chapter the right length?
Should anything missing be added?
Can anything superfluous be removed (e.g. by deleting some section that does not work so well or  by using less jargon, less formulae, lees diagrams, less references).?
What are the aspects of the chapter that authors  SHOULD  change?

_The chapter is a reasonable length. There are a lot of messages at the end of the chapter. The message about the forest and trees wasn't clearly tied to the rest of the chapter as the meaning of 'data' is not overly clear. Similarly the message about junior researchers comes out of the blue and doesn't tie into the chapter._

## Gotta Mantra?

We encouraged (but did not require) the chapter title to be a mantra or something cute/catchy, i.e., some slogan reflecting best practice for data science for SE?  If you have suggestion for a better title, please put them here.

_I am horrible at titles. Maybe play on the inappropriate assumptions more in the title?_

## Best Points

What are the best points of the chapter that the authors should NOT change?

_Really like the 'Some Reflections' portion. Like having a bullet list of actionable take-aways at the end of the chapter._

**Comment by baishakhir** [comment]
## Title of chapter

Are we really helping developers?

## URL to the chapter

https://github.com/ds4se/chapters/blob/master/alexorso/are-we-helping-developers.md

## Message?

No matter how popular a research sub-field in software engineering becomes, it should be evaluated with users/developers to investigate whether the proposed approaches are really helping them. 

## Accessible?

Overall, the chapter is pretty accessible to the general audience. The section header “With Big Data Comes Great Responsibility” is a bit ambiguous w.r.t. its text. It would be better to describe what "Big Data" actually means here and what kind of responsibilities authors are expecting from the practitioners.

Some Typos:

Section Premise: findings related “to” the latter  --- ‘to’ is missing
RQ1: extra will
Section Assessing the Usefulness of a Technique,  last para: stageatistical will be statistical
Section Summary, first para: gnerated -> generated

## Size?

The size is fine. I find the first paragraph in section 1 and first paragraph in section 2 a little repetitive, especially when the authors are explaining statistical bug localization. A little reorganization might be better.

## Gotta Mantra?

The chapter titling “Are we really helping developers" is actually about “Are we really helping developers with Statistical Fault Localization”.  In the premise section, the authors can make it explicit that they are presenting their arguments with a  case study of  Statistical Fault Localization.

## Best Points

Reflection Section. I really like the discussion about the initial assumptions and reality.

**Comment by lauriew** [comment]
@alexorso 

You have the two reviews.  As was previously sent out, we were aiming for people to revise their chapters by January 13.  I know the time is relatively short, but since the chapter is not that long it should hopefully not take too long.

Here's my comments when I read in addition to those from the other reviewers:
I found the "Premise" section to be defensive, which seems a tough way to start out a chapter.  Perhaps the word "argue" is what made me feel that way.

Instead of using "Reference [1]" was awarded ... you could say "Jones et al. [1] was awarded"  to avoid using the reference as the subject of the sentence.  (same with using Reference [2] later in the sentence).

I found this sentence hard to parse "The results we discussed above provide evidence that statistical fault localization, a technique that appears to work well on paper, may not work as well, and thus help developers as expected, in practice."  I feel like it would be more clear if you said "and thus may not [actually?] help developers in practice".

"Find causes instead OF correlations"

I agree the title is pretty general while the chapter is on statistical fault localization.  I know we asked for a catchy title.  I think the title may hang together if you bring in more of the message that with all research we should check with the users to see if we are really helping them before the research goes on and on ... and giving statistical fault localization as an **example** of the dangers when no one ever checks in with the user.

**Comment by alexorso** [comment]
Many thanks to Baishakhi, Gail, and Laurie for their feedback, which made me realize that I was trying to fit too many (sometimes unrelated) concepts into this chapter. I extensively revised the text based on their comments and tried to make it more focused. I hope the different parts are better tied together in this new version.

**Comment by lauriew** [comment]
@alexorso Thanks Alex, looks good.
